acceptance/tests/direct_puppet/supports_utf8.rb

```diff
@@ -50,18 +53,18 @@
   step 'run agent(s)' do
     with_puppet_running_on(master, {}) do
       agents.each do |agent|
-        fqdn = agent.hostname
+        fqdn = agent_to_fqdn(agent)
         config_version = ''
         config_version_matcher = /configuration version '(\d+)'/
-        on(agent, puppet("agent -t --environment #{tmp_environment} --server #{master.hostname}"),
+        on(agent, puppet("agent -t --environment '#{tmp_environment} --server #{master.hostname}'"),
            :acceptable_exit_codes => 2).stdout do |agent_out|
           config_version = agent_out.match(config_version_matcher)
         end
         on(agent, "cat #{tmp_file[fqdn]}").stdout do |result|
           assert_equal(file_contents, result, 'file contents did not match accepted')
         end
         on(agent, "rm #{tmp_file[fqdn]}", :accept_all_exit_codes => true)
-        on(agent, puppet("agent -t --environment #{tmp_environment} --server #{master.hostname} --use_cached_catalog"),
+        on(agent, puppet("agent -t --environment '#{tmp_environment}' --server #{master.hostname} --use_cached_catalog"),
            :acceptable_exit_codes => 2).stdout do |agent_out|
           assert_match(config_version_matcher, result, 'agent did not use cached catalog')
```

Comment:
I know that you didn't change this, but where is "result" for this assert_match coming from?

Should it be "agent_out" instead?